workers_test.go

```diff
@@ -14,53 +14,58 @@ import (
 )
 
 const (
-	workerCount   = 1000
+	workerCount   = 100000
 	workerTimeout = time.Millisecond * 300
 	runTimes      = 100000
 )
 
 type WorkerOne struct {
+	Count int
+	sync.Mutex
 }
 type WorkerTwo struct {
+	Count int
+	sync.Mutex
 }
 
-func NewWorkerOne() Worker {
+func NewWorkerOne() *WorkerOne {
 	return &WorkerOne{}
 }
 
-func NewWorkerTwo() Worker {
+func NewWorkerTwo() *WorkerTwo {
 	return &WorkerTwo{}
 }
 
+func (wo *WorkerOne) CurrentCount() int {
+	wo.Lock()
+	defer wo.Unlock()
```

Comment:
Sorry my review is coming late, but for next time I suggest just use the atomic package to increment counters concurrently! 👍🏼